tests/SearchVariantSubsitesTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+class SearchVariantSubsiteTest extends SapphireTest
+{
+
+    private static $index = null;
+
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        // Check versioned available
+        if (!class_exists('Subsite')) {
+            return $this->markTestSkipped('The versioned decorator is not installed');
+        }
+
+        if (self::$index === null) {
+            self::$index = singleton('SearchVariantSubsiteTest');
+        }
+
+        SearchUpdater::bind_manipulation_capture();
+
+        Config::inst()->update('Injector', 'SearchUpdateProcessor', array(
+            'class' => 'SearchUpdateImmediateProcessor'
+        ));
+
+        FullTextSearch::force_index_list(self::$index);
+        SearchUpdater::clear_dirty_indexes();
+    }
+
+    public function testQueryIsAlteredWhenSubsiteNotSet()
+    {
+        $index = new SolrIndexTest_FakeIndex();
+        $query = new SearchQuery();
+
+        //typical behaviour: nobody is explicitly filtering on subsite, so the search variant adds a filter to the query
+        $this->assertTrue(empty($query->require['_subsite']));
+        $variant = new SearchVariantSubsites();
+        $variant->alterDefinition('SearchUpdaterTest_Container', $index);
+        $variant->alterQuery($query, $index);
+
+        //check that the "default" query has been put in place: it's not empty, and we're searching on Subsite ID:0 and
+        // an object of SearchQuery::missing
+        $this->assertFalse(empty($query->require['_subsite']));
+        $this->assertTrue($query->require['_subsite'][0] == 0);
+        $this->assertTrue($query->require['_subsite'][1] instanceof SearchQuery::$missing);
+    }
+
+
+    public function testQueryIsAlteredWhenSubsiteIsSet()
+    {
+        //now we want to test if somebody has already applied the _subsite filter to the query
+        $index = new SolrIndexTest_FakeIndex();
+        $query = new SearchQuery();
+
+        //check that _subsite is not applied yet
+        $this->assertTrue(empty($query->require['_subsite']));
```

Comment:
`$this->assertEmpty($query->require['_subsite'])` Although, if the key may not exist then maybe you'll have to leave it as is - please add a comment if that's the case.